extralit-server/src/extralit_server/contexts/ocr/text.py

```diff
@@ -12,3 +12,251 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""
+PDF text extraction using PyMuPDF service with RQ integration.
+
+This module provides PDF text extraction functionality with both RQ-based
+asynchronous processing and HTTP fallback for reliability. It follows the
+senior developer approach of simple, reusable functions with comprehensive
+error handling and fallback mechanisms.
+"""
+
+import asyncio
+import logging
+import os
+from typing import Any, Optional
+
+import httpx
+
+from .rq_client import cancel_job, enqueue_pdf_extraction, get_job_status, is_redis_available
```

Comment:
FYI python imports should be using the full path starting with extralit_server, rather than using relative imports. Even though it requires creating the pyproject.toml (or setup.py) to setup the python package, it makes it a lot hard to make a mistake and having your imports break.